spark/src/test/scala/org/apache/spark/sql/delta/skipping/ClusteredTableTestUtils.scala

```diff
@@ -60,6 +61,33 @@ trait ClusteredTableTestUtilsBase extends SparkFunSuite with SharedSparkSession
     assert(expectedClusteringColumns == actualClusteringColumns)
   }
 
+  // Verify the operation parameters of the last history event contains `clusterBy`.
+  protected def verifyDescribeHistoryOperationParameters(
+      table: String
+  ): Unit = {
+    val deltaLog = if (table.startsWith("tahoe.") || table.startsWith("delta.")) {
+      // Path based table.
+      DeltaLog.forTable(spark, table.drop(6).replace("`", ""))
+    } else {
+      DeltaLog.forTable(spark, TableIdentifier(table))
+    }
+    val clusteringColumns =
+      ClusteringColumnInfo.extractLogicalNames(deltaLog.unsafeVolatileSnapshot)
```

Comment:
Can this `deltaLog.unsafeVolatileSnapshot` be avoided if you do `DeltaLog.forTableWithSnapshot()` above?

Reply:
good idea, updated!